app/services/hyrax/solr_query_service.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+module Hyrax
+  ##
+  # Methods in this class are providing functionality previously supported by ActiveFedora::SolrQueryBuilder.
+  # It includes methods to build and execute a query.
+  class SolrQueryService
+    attr_reader :query, :solr_service
+
+    def initialize(query: [], solr_service: Hyrax::SolrService)
+      @query = query
+      @solr_service = solr_service
+    end
+
+    ##
+    # @return [Hash] the results returned from solr for the current query
+    def get
+      solr_service.get(build)
+    end
+
+    ##
+    # @return [Integer] the number of results that match the query in solr
+    def count
+      solr_service.count(build)
+    end
+
+    ##
+    # @return [String] the combined query that can be submitted to solr
+    def build
+      return 'id:NEVER_USE_THIS_ID' if @query.blank? # forces this method to always return a valid solr query
+      @query.join(' AND ')
+    end
+
+    ##
+    # @return [Hyrax::SolrQueryService] the existing service with the query reset to empty
+    def reset
+      @query = []
+      self
+    end
+
+    ##
+    # @param ids [Array] id_array the ids that you want included in the query
+    # @return [Hyrax::SolrQueryService] the existing service with id query appended
+    def with_ids(ids: [])
+      raise ArgumentError, "Expected [Array] for ids:, but received <#{ids.class}>" unless ids.is_a? Array
```

Comment:
it looks like this will reject a lot of arguments that would otherwise work fine? is there any harm in dropping the type check?